### PR TITLE
Feature/article api integration

### PR DESCRIPTION
**Pull request by eliotmess**
I implemented only logic and API integration in this one.

Next I will add displaying of data and styling.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,26 +17,36 @@
...
     "react-intl": "3.9.1",
     "react-native": "0.61.5",
     "react-native-config": "0.12.0",
+    "react-native-dotenv": "0.2.0",
     "react-native-gesture-handler": "1.5.2",
     "react-navigation": "4.0.10",
     "react-navigation-stack": "1.10.3",
     "react-redux": "7.1.3",
     "redux": "4.0.4",
+    "redux-logger": "3.0.6",
```
Comment (lukaszblasz) [inline package.json]: redux logger shoudl be in dev-dependecies not standard dependecies. Same all typescript stuff like ciustom types etc.

File: .env [hunk 2]
```diff
@@ -1,2 +1,4 @@
 TYPE=development
-API_HOST=https://api.staging.sample.com
\ No newline at end of file
+API_HOST=https://api.staging.sample.com
```
Comment (adamkosmala) [inline .env]: what's the purpose of API_HOST in this case, since we have API_URL?

File: android/.project [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (adamkosmala) [inline android/.project]: correct me if I'm wrong, but `.project` files should be in gitignore

Comment (adamkosmala) [reply android/.project]: same for `perfs` files

File: src/app/services/articles-service/index.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import dayjs from 'dayjs';
+// eslint-disable-next-line
+// @ts-ignore
```
Comment (lukaszblasz) [inline src/app/services/articles-service/index.ts]: ts ignore?

File: src/app/modules/MainArticlesScreen/MainArticlesScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+import { NavigationStackScreenProps } from 'react-navigation-stack';
+import { WrappedComponentProps } from 'react-intl';
+import { v1 } from 'uuid';
+
+import {
+  ArticleData,
+  topicValues,
+  timeValues,
+  sortValues,
+} from '@modules/MainArticlesScreen/namespace';
```
Comment (adamkosmala) [inline src/app/modules/MainArticlesScreen/MainArticlesScreen.tsx]: why modules, not pages, like we have in web version of CRA? let's keep things consistent, please

File: src/app/services/articles-service/index.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import { API_KEY as apiKey } from 'react-native-dotenv';
+import ApiService from '@services/config';
+import {
+  topicValues,
+  sortValues,
+  timeValues,
+} from '@modules/MainArticlesScreen/namespace/index';
+
+const now = dayjs();
```
Comment (lukaszblasz) [inline src/app/services/articles-service/index.ts]: Should now be here? It get now() when js is loaded for the first time at the beggining, not when specific service is called. So if you open application and leave it for hour or 2, and call the servioce the now will point to the time before current time.

File: src/app/services/articles-service/index.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      from = now.subtract(7, 'day').format('YYYY-MM-DD');
+      break;
+    case timeValues.month:
+      from = now.subtract(10, 'day').format('YYYY-MM-DD');
+      break;
+    default:
+      from = to;
+  }
+
+  const myUrl = `everything?q=${selectedTopic}&page=${page}&sortBy=${sort}&from=${from}&to=${to}&apiKey=${apiKey}`;
```
Comment (lukaszblasz) [inline src/app/services/articles-service/index.ts]: Is no other way to build query parameters? Using axios wne you specify params as js object they are automaticly extracted as quesryParameters. I thing we have the same problem in our standard react app.

File: .env (hunk 2 above)
Comment (eliotmess) [reply .env]: none I guess. It's a leftover and I'll delete it.

File: android/.project (hunk 3 above)
Comment (eliotmess) [reply android/.project]: I found out that `.project` probably should be ignored ([gitignore.io](https://www.gitignore.io/api/reactnative,androidstudio) and [react native's repo](https://github.com/facebook/react-native/blob/master/.gitignore)).

No info about `perfs` files though.

File: src/app/services/articles-service/index.ts (hunk 4 above)
Comment (eliotmess) [reply src/app/services/articles-service/index.ts]: It's a workaround of `react-native-dotenv` incompatibility with Typescript.
I've provided an explanation but in `config.ts` file below.

File: src/app/store/RootReducer.ts [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import { combineReducers, Reducer } from 'redux';
 
-import { mainScreenReducer } from '@modules/MainArticlesScreen/store/reducers';
+// import { mainScreenReducer } from '@pages/MainArticlesScreen/store/reducers';
```
Comment (adamkosmala) [inline src/app/store/RootReducer.ts]: commented out code

File: src/app/services/articles-service/index.ts [hunk 9]
```diff
@@ -7,16 +7,15 @@ import {
...
-const now = dayjs();
+} from '@pages/MainArticlesScreen/namespace/index';
 
 const getArticlesService = (
   page = 1,
   topic?: string,
   sortBy?: string,
   date?: string,
 ) => {
+  const now = dayjs();
```
Comment (adamkosmala) [inline src/app/services/articles-service/index.ts]: if `now` is not reused anywhere, can we simply keep `const to dayjs().format(...)`?

**Pull request merged**